mats/io.ms

```diff
@@ -2135,6 +2144,166 @@
         (= q (custom-port-buffer-size)))))
 )
 
+(mat transcoded-port-buffer-size
+  (let ([x (transcoded-port-buffer-size)])
+    (and (fixnum? x) (> x 0)))
+  (error? (transcoded-port-buffer-size 1024 15))
+  (error? (transcoded-port-buffer-size 'shoe))
+  (error? (transcoded-port-buffer-size 0))
+  (error? (transcoded-port-buffer-size -15))
+  (error? (transcoded-port-buffer-size (+ (most-positive-fixnum) 1)))
+  (error? (transcoded-port-buffer-size 1024.0))
+  (parameterize ([transcoded-port-buffer-size (* (transcoded-port-buffer-size) 2)])
+    (let ([ip (transcoded-port (open-bytevector-input-port (string->utf8 "howdy"))
+                (make-transcoder (utf-8-codec)))])
+      (= (transcoded-port-buffer-size)
+         (string-length (port-input-buffer ip)))))
+  (let ()
+    ;; Codecs used to rely on transcoded port buffer size being <= codec buffer
+    ;; size when constructing ioffsets for transcoded input port. Check that we
+    ;; no longer make that assumption.
+    (define (check-buffers codec-buf-size tp-buf-size)
+      (define (check-codec codec)
+        (let* ([tx (make-transcoder (codec))]
+               [src-bv
+                (string->bytevector
+                 ;; Chez Scheme handles Unicode string literals just fine, but
+                 ;; the mats generate prettytest.ss from sources including this
+                 ;; file and clause 1 of the string-ports mat fails if we
+                 ;; include exotic characters here, so use escape characters.
+                 (string-append
+                  "abcde"
+                  ;; utf-8 2-byte
+                  "\x3B1;\x3B2;\x3B3;\x3B4;\x3B5;"
+                  ;; utf-8 3-byte
+                  "\x24B6;\x24B7;\x24B8;\x24B9;\x24BA;\x24BB;"
+                  ;; utf-8 4-byte
+                  "\x1F311;\x1F312;\x1F313;\x1F314;\x1F315;\x1F316;\x1F317;\x1F318;")
+                 tx)]
+               [min-codec-buffer-size 4]
+               [bp
+                (parameterize ([make-codec-buffer
+                                (lambda (bp)
+                                  (make-bytevector
+                                   (max min-codec-buffer-size codec-buffer-size)))])
```

Comment:
This function is never called, which is good since `codec-buffer-size` is unbound.  (It's spelled `codec-buf-size` here.)

Is this supposed to be part of the `parameterize` in the next binding in this `let*` (which actually does have a call to `transcoded-port` in its body)?

Reply:
Indeed it is. Thanks for spotting this. Force pushing a fix for this and the next one.